frontend/src/api/k8s/storageClasses.ts

```diff
@@ -1,9 +1,57 @@
-import { k8sListResource } from '@openshift/dynamic-plugin-sdk-utils';
-import { StorageClassKind } from '~/k8sTypes';
+import {
+  k8sGetResource,
+  k8sListResource,
+  k8sPatchResource,
+  Patch,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import { K8sAPIOptions, StorageClassConfig, StorageClassKind } from '~/k8sTypes';
 import { StorageClassModel } from '~/api/models';
+import { applyK8sAPIOptions } from '~/api/apiMergeUtils';
 
 export const getStorageClasses = (): Promise<StorageClassKind[]> =>
   k8sListResource<StorageClassKind>({
     model: StorageClassModel,
     queryOptions: {},
   }).then((listResource) => listResource.items);
+
+export const getStorageClass = (name: string): Promise<StorageClassKind> =>
+  k8sGetResource<StorageClassKind>({
+    model: StorageClassModel,
+    queryOptions: { name },
+  });
+
+export const updateStorageClassConfig = async (
+  name: string,
+  config: Partial<Omit<StorageClassConfig, 'lastModified'>> | undefined,
```

Comment:
There is no need to use both `Partial` and `Omit` here IMO, the `Partial` has already made all the props in the type to be optional.